manifests/pulp.pp

```diff
@@ -0,0 +1,201 @@
+  Integer[1] $puppet_wsgi_processes = 1,
+  Boolean $enable_docker = true,
+  Boolean $enable_deb = false,
+
+  Stdlib::Host $broker_host = 'localhost',
+  Stdlib::Port $broker_port = 5671,
+
+  Optional[String] $default_password = undef,
+  Optional[String] $yum_max_speed = undef,
+  Optional[Integer[1]] $num_workers = undef,
+  Optional[Integer[0]] $worker_timeout = undef,
+
+  Optional[Stdlib::Host] $proxy_host = undef,
+  Optional[Stdlib::Port] $proxy_port = undef,
+  Optional[String] $proxy_username = undef,
+  Optional[String] $proxy_password = undef,
+
+  Optional[Stdlib::Fqdn] $server_name = undef,
+  Optional[String] $ssl_protocol = undef,
+
+  String $mongodb_name = 'pulp_database',
+  Optional[String] $mongodb_seeds = undef,
+  String $mongodb_username = undef,
+  String $mongodb_password = undef,
+  String $mongodb_replica_set = undef,
+  Boolean $mongodb_ssl = false,
+  Optional[Stdlib::Absolutepath] $mongodb_ssl_keyfile = undef,
+  Optional[Stdlib::Absolutepath] $mongodb_ssl_certfile = undef,
+  Boolean $mongodb_verify_ssl = true,
+  Optional[Stdlib::Absolutepath] $mongodb_ca_path = undef,
+  Boolean $mongodb_unsafe_autoretry = false,
+  Optional[Enum['majority', 'all']] $mongodb_write_concern = undef,
+  Boolean $mongodb_manage_db = true,
+) {
+
+  $pulp2 = $enable_ostree or $enable_yum or $enable_file or $enable_puppet or $enable_docker or $enable_deb
+  $pulp3 = false
+
+  include certs::apache
+
+  if $pulp2 {
+    include certs::qpid_client
+
+    class { 'pulp':
+      server_name            => pick($server_name, $certs::apache::hostname),
+      ca_cert                => $certs::apache::ca_cert,
+      https_cert             => $certs::apache::apache_cert,
+      https_key              => $certs::apache::apache_key,
+      db_ca_path             => $mongodb_ca_path,
+      db_name                => $mongodb_name,
+      db_password            => $mongodb_password,
+      db_replica_set         => $mongodb_replica_set,
+      db_seeds               => $mongodb_seeds,
+      db_ssl                 => $mongodb_ssl,
+      db_ssl_certfile        => $mongodb_ssl_certfile,
+      db_ssl_keyfile         => $mongodb_ssl_keyfile,
+      db_unsafe_autoretry    => $mongodb_unsafe_autoretry,
+      db_username            => $mongodb_username,
+      db_verify_ssl          => $mongodb_verify_ssl,
+      db_write_concern       => $mongodb_write_concern,
+      default_password       => $default_password,
+      enable_crane           => true, # requires https://github.com/theforeman/puppet-pulp/pull/378
+      enable_deb             => $enable_deb,
+      enable_docker          => $enable_docker,
+      enable_iso             => $enable_file,
+      enable_katello         => !$is_mirror,
+      enable_ostree          => $enable_ostree,
+      enable_puppet          => $enable_puppet,
+      enable_rpm             => $enable_yum,
+      manage_db              => $mongodb_manage_db,
+      manage_httpd           => $is_mirror,
```

Comment:
I assume `$is_mirror` has value `true` for the proxy scenario and `false` for katello. Can I take the same approach when building the logic for pulp3?

Reply:
It is sort of based on Pulp 3 where there's no longer the concept of a master and a node. Just the `is_mirror` parameter. I've started to like that.